presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -111,13 +111,13 @@ public ReorderJoins(CostComparator costComparator)
     @Override
     public boolean isEnabled(Session session)
     {
-        return getJoinReorderingStrategy(session) == COST_BASED;
+        return getJoinReorderingStrategy(session) == COST_BASED && getMaxReorderedJoins(session) > 0;
```

Comment:
remove `&& getMaxReorderedJoins(session) > 0;` since the `@Min(2)` check

Reply:
and ![onemore](https://media.tenor.co/images/7584cf5e881cd1dc603495abb1037f42/tenor.gif)